director/director.go

```diff
@@ -1156,15 +1156,33 @@ func getHealthTestFile(ctx *gin.Context) {
 	}
 }
 
+func collectClientVersionMiddleware(c *gin.Context) {
```

Comment:
Why add this to a middleware instead of collecting the metric directly in `redirectToCache` and `redirectToOrigin`? My concern is that by the `ShortcutMiddleware` route we have set up in `launchers/director_serve.go` is what gets hit in the majority of client requests (but not cache requests), and it looks like that'll bypass this middleware.